### DISCUSSION TITLE
Is it possible to convert the audio from a video call meeting using python Bot?

### DISCUSSION THREAD

**Discussion by samueljazzjohn** [deepgram/community#253] [General help] [answered]
I need to chunk the audio data and then convert it to text. Is that possible using Deepgram? Is that possible to get the audio directly from the deepgram Python SDK?

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @samueljazzjohn, Deepgram offers both streaming and pre-recorded transcription, depending on whether you need the transcript in real time or at the end of a call.

You would submit either the recording itself or a URL to the recording, and Deepgram returns the JSON transcript. Our Nova model is particularly fast, so if you don't need real time, my recommendation would be to submit only the audio of the meeting to Deepgram, and we could return the resulting transcript in about 12 seconds for an hour of audio. This is all feasible and a standard use case of the product and can be done with the Deepgram Python SDK. You would use the `deepgram.transcription.sync_prerecorded()` method. 

See the [Python SDK on GitHub](https://github.com/deepgram/deepgram-python-sdk) and our [pre-recorded](https://developers.deepgram.com/docs/python-sdk-pre-recorded-transcription) and [streaming](https://developers.deepgram.com/docs/python-sdk-streaming-transcription) getting-started guides in Python.

**Comment by samueljazzjohn** [comment]
I would like to receive the transcription of the audio in chunks of 5 seconds. In other words, I want to divide the audio into 5-second segments and obtain the corresponding transcriptions for each segment. Is that possible. I think Deepgram chunking is based on the pause during the communication. Am I right?

**Reply by nikolawhallon** [reply]
Deepgram has its own chunker to optimize a balance between transcription quality and latency of returned results, if chunking strictly every 5 seconds, it's likely that the edges of those chunks will cut through spoken words, and then the model will not be able to transcribe those words at the boundaries very well.

For streaming, by default Deepgram will find an optimal cut/chunk point every 3-5 seconds, unless it detects the end of someone's speech (determined by 10ms of silence following someone's on-going speech), so if this roughly works for your 5 second requirement, then that's great, if not, can you explain a bit more the use-case to see if maybe we have some configuration that could help better?

**Reply by jkroll-deepgram** [reply]
Thanks @nikolawhallon! Nearly all applications can be designed independent of the chunk size sent to and returned by Deepgram. 

Adding on, we have another feature that could be of interest, [interim results](https://developers.deepgram.com/docs/interim-results), where you can get our best prediction of speech every ~1-2 seconds, with a finalized prediction every ~3-7 seconds. We don't offer the option of explicitly requesting audio chunked every N seconds, as the prediction depends on the surrounding context of pauses, speaker changes, etc. But if you do use interim results, you can then construct a transcript for each 5-second chunk if you need it.

Seconding that we can advise more broadly if you'd like to share more about how you're planning to use these chunked transcriptions as well.
